### PR TITLE
front: fix conflicts card displayed time

### LINKED ISSUE #9454: front: the times in the conflict cards are incorrect
### What happened?

I created a train which leaves at 12h. I duplicated, the 2nd train also leaves at 12h.
I should see a conflict starting at 12h, but it starts at 10h:
Image

### What did you expect to happen?

I should see a conflict starting at 12h.

### How can we reproduce it (as minimally and precisely as possible)?

1. Create a scenario
2. Create a train, which leaves at 12h
3. Create a second train without changing any inputs.
4. Display the simulation results, open the conflicts tab

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Firefox

### OSRD version (top right corner `Account` button > `Informations`)

.

### PR DESCRIPTION
**Pull request by Akctarus**
closes #9454

### CODE REVIEW

File: front/src/utils/date.ts [hunk 1]
```diff
@@ -20,7 +20,10 @@ const userTimeZone = dayjs.tz.guess(); // Format : 'Europe/Paris'
  * @param dateTimeString date string in ISO format
  * @returns string "HH:MM:SS"
  */
-export function extractHHMMSS(dateTimeString?: string) {
+export const isoStringToLocalDate = (dateString: string) =>
```
Comment (emersion) [inline front/src/utils/date.ts]: Hm, it's not super clear from the name that this function is just formatting the hours+minutes+seconds, and leaving out the day+month+year.

Also other call-sites may need slightly different formatting (e.g. leaving out the seconds), so maybe we should move this inside `ConflictCard` for now?

**Comment by emersion** [comment]
A good follow-up task may to be get rid of `extractHHMMSS()` completely, because that function is just a footgun.

**Comment by Akctarus** [comment]
> A good follow-up task may to be get rid of `extractHHMMSS()` completely, because that function is just a footgun.

Yes, I'd thought of doing it in the refactor ticket for the date.ts file, there's a lot to do there.

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested. Agreed with @emersion, maybe the name of the function could be better

**Review by Uriel-Sautron — APPROVED** [review]
LGTM, tested

File: front/src/utils/date.ts (hunk 1 above)
Comment (Akctarus) [reply front/src/utils/date.ts]: Fixed :)

File: front/src/utils/date.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ const userTimeZone = dayjs.tz.guess(); // Format : 'Europe/Paris'
  * @param dateTimeString date string in ISO format
  * @returns string "HH:MM:SS"
  */
-export function extractHHMMSS(dateTimeString?: string) {
+function extractHHMMSS(dateTimeString?: string) {
```
Comment (emersion) [inline front/src/utils/date.ts:23]: Oh, so is this function unused now? If so, can we nuke it?

Comment (Akctarus) [reply front/src/utils/date.ts:23]: it's used in timestampToHHMMSS, so no need to export it anymore, but we need to keep it until we refactor the file

Comment (emersion) [reply front/src/utils/date.ts:23]: Ah right, makes sense! Good to stop exporting it anyways :)

**Review by emersion — APPROVED** [review]
LGTM!

**Pull request merged**